File: 백준/Silver/1463. 1로 만들기/1로 만들기.py
a=int(input())
dp = []
for i in range(0,1000002):
    dp.append(0)
for i in range(2,a+1):
    dp[i]=dp[i-1]+1

    if i%2==0:
        dp[i]=min(dp[i//2]+1,dp[i])
    if i%3==0:
        dp[i]=min(dp[i//3]+1,dp[i])

print(dp[a])

File: 백준/Silver/1463. 1로 만들기/1로 만들기.java
import java.util.Scanner;
public class Main {
public static void main(String []args) {
	Scanner sc=new Scanner(System.in);
	int[] dp=new int[1000001];
	int n=sc.nextInt();
	dp[0]=0;
	dp[1]=0;
	for(int i=2;i<=n;i++) {
		dp[i]=dp[i-1]+1;
		if(i%2==0)
			dp[i]=Math.min(dp[i/2]+1, dp[i]);
		if(i%3==0)
			dp[i]=Math.min(dp[i/3]+1, dp[i]);
	}
	System.out.print(dp[n]);
    }
}
